OptimizelySDK.Tests/NotificationTests/NotificationCenterTests.cs

```diff
@@ -0,0 +1,195 @@
+﻿/* 
+ * Copyright 2017, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Moq;
+using NUnit.Framework;
+using OptimizelySDK.Entity;
+using OptimizelySDK.ErrorHandler;
+using OptimizelySDK.Event;
+using OptimizelySDK.Logger;
+using OptimizelySDK.Notifications;
+
+namespace OptimizelySDK.Tests.NotificationTests
+{
+    public class NotificationCenterTests
+    {
+        private Mock<ILogger> LoggerMock;
+        private NotificationCenter NotificationCenter;
+        private TestNotificationCallbacks TestNotificationCallbacks;
+
+        [SetUp]
+        public void Setup()
+        {
+            LoggerMock = new Mock<ILogger>();
+            LoggerMock.Setup(i => i.Log(It.IsAny<LogLevel>(), It.IsAny<string>()));
+
+            NotificationCenter = new NotificationCenter(LoggerMock.Object);
+            TestNotificationCallbacks = new TestNotificationCallbacks();
+        }
+
+        [Test]
+        public void TestAddAndRemoveNotificationListener()
+        {
+            var notificationType = NotificationCenter.NotificationType.Decision;
+
+            // Verify that callback added successfully.
+            Assert.AreEqual(1, NotificationCenter.AddNotification(notificationType, TestNotificationCallbacks.TestDecisionCallback));
+            Assert.AreEqual(1, NotificationCenter.NotificationsCount);
+
+            // Verify that callback removed successfully.
+            Assert.AreEqual(true, NotificationCenter.RemoveNotification(1));
+            Assert.AreEqual(0, NotificationCenter.NotificationsCount);
+        }
+
+        [Test]
+        public void TestAddMultipleNotificationListeners()
+        {
+            var notificationTypeDecision = NotificationCenter.NotificationType.Decision;
```

Comment:
i assume, if these are being used again and again, please take it as a global variable rather to define in each variable for common purpose. notification type of decision, track...

Reply:
Agreed. Will do that in the morning.

Reply:
Done